### ISSUE TITLE
Error page shown on delete although worked.

### ISSUE THREAD

**Issue by ulpaulpa** [barryf/micropublish#70]
I'm using my own server implementation (https://codeberg.org/ulpa/minibar). It response with http 201, a location header and an empty body. The tests on micropub.rocks pass, so there shouldn't be any issues with the server.

**Comment by barryf** [comment]
Thanks for the report, @ulpaulpa. What was the error message you received? Did you get any information in the request to your Micropub server that may help us debug?

**Comment by ulpaulpa** [comment]
The server didn't return any error message and the post was deleted, but micropublish said there was an error.

**Comment by ulpaulpa** [comment]
```There was an error making a request to your Micropub endpoint. The error received was: ```
That's the error message. (nothing behind the colon)

**Comment by barryf** [comment]
I had a look at the logs and this may be the culprit. 

The URL that's being passed is `<class 'str'> https://ulpaulpa.de/p/testtest-1` which is invalid.

**Comment by ulpaulpa** [comment]
I wanted to ask that too, but that's when creating posts. When I'm deleting a post the post gets deleted, but micropublish says there was an error

**Comment by barryf** [comment]
I've found a few error responses from your server. 

At 15:21 UTC there was no message but I note the URL `https://ulpaulpa.de/p/-3/` is currently `403 Forbidden`.

15:16 UTC:
```
<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

11:39 UTC: 
```
File "/home/ulpa/blog/micropub-server/write.py", line 94, in <module>
    os.remove("/home/ulpa/html/"+path+"/index.html")
FileNotFoundError: [Errno 2] No such file or directory: '/home/ulpa/html/p/sluggy/index.html'
```

**Comment by ulpaulpa** [comment]
1. I deleted it (the folder still exists, the index.html file not)
2. I forgot to restart the server
3. I tried to redelete the post, because of the error message.

**Comment by barryf** [comment]
I think I've found it. For a delete Micropublish is expecting a 200 or 204 but as you stated at the top you're returning a 201 with a Location. Is there a reason you want to specify a Location when deleting?

**Comment by ulpaulpa** [comment]
No, but I return 201 for all successful post requests and I don't want to add an exception just for deletes as its allowed: https://www.w3.org/TR/micropub/#response-2

**Comment by barryf** [comment]
Micropublish accepts 201s for an undelete but not a delete. I think that paragraph in the spec could have been a little clearer, but the key part is "Otherwise, the server must respond with 200 or 204."
